src/main/java/org/jnode/fs/xfs/XfsEntry.java

```diff
@@ -72,7 +70,7 @@ else if (mode == UnixFSConstants.S_IFDIR)
             return AbstractFSEntry.DIR_ENTRY;
         else if (mode == UnixFSConstants.S_IFREG || mode == UnixFSConstants.S_IFLNK ||
                 mode == UnixFSConstants.S_IFIFO || mode == UnixFSConstants.S_IFCHR ||
-                mode == UnixFSConstants.S_IFBLK)
+                mode == UnixFSConstants.S_IFBLK || mode == UnixFSConstants.S_IFSOCK)
```

Comment:
It's a bug found when we run this code with engine, and the socket one will simply return false as we didn't consider it as a file previously.